### DISCUSSION TITLE
Capturing an error before the source Filter script is executed

### DISCUSSION THREAD

**Discussion by JonEBee** [nextgenhealthcare/connect#4793] [Q&A] [answered]
Mirth Connect 3.8.1 on RHEL 7.9

Saw the following error information:
```Transformer error
ERROR MESSAGE: Error evaluating transformer
com.mirth.connect.server.MirthJavascriptTransformerException: 
CHANNEL:	AHM_MEP_GetMemberMessages_QA
CONNECTOR:	sourceConnector
SCRIPT SOURCE:	
SOURCE CODE:	
2686:         }
2687:         eval('importPackage(' + Packages.java.lang.Class.forName(className).getPackage().getName() + ')');
2688:     }
2689: }
2690: function doScript() {
2691: msg = JSON.parse(connectorMessage.getTransformedData());
2692: function filterRule1() {
2693: //
2694:      var logString = "";
2695: 
LINE NUMBER:	2691
DETAILS:	SyntaxError: Missing comma in object literal
```

The cause was the lack of a comma after one of the properties in middle of the JSON doc. The culprit line, 2691, is the last line before the first Source Filter rule is run.

Is there some way to capture that error, similar to the try...catch JS mechanism, in order to return a more appropriate response, such as a more detailed API response?

Thanks in advance for your response.

**Comment by pacmano1** [comment, marked as answer]
Yup, set the data type to raw on your inbound listener and `JSON.parse` msg in a try..catch.
